src/execution/compiler/operator/insert_translator.cpp

```diff
@@ -1 +1,111 @@
-namespace terrier::execution::compiler {}  // namespace terrier::execution::compiler
+
+#include <execution/compiler/operator/insert_translator.h>
+#include <execution/compiler/translator_factory.h>
+#include <execution/compiler/storage/pr_filler.h>
+
+#include "execution/compiler/operator/insert_translator.h"
+#include "execution/compiler/function_builder.h"
+
+namespace terrier::execution::compiler {
+InsertTranslator::InsertTranslator(const terrier::planner::InsertPlanNode *op, CodeGen *codegen)
+    : OperatorTranslator(codegen),
+      op_(op),
+      inserter_struct_(codegen->NewIdentifier(inserter_name_)),
+      table_pr_(codegen->NewIdentifier(table_pr_name_)),
+      table_schema_(codegen->Accessor()->GetSchema(op_->GetTableOid())),
+      table_pm_(codegen->Accessor()->GetTable(op_->GetTableOid())->ProjectionMapForOids(op_->GetParameterInfo())) {}
+
+
+void InsertTranslator::Produce(FunctionBuilder *builder) {
+  // generate the code for the insertion
```

Comment:
Break this function into multiple function calls. Otherwise it's hard to read.
See other translator for an idea on how to break things down.